server/internal/server/server.go

```diff
@@ -30,13 +33,19 @@ type cacheGetter interface {
 }
 
 type tokenIntrospector interface {
-	TokenIntrospect(token string) (*dex.Introspection, error)
+	TokenIntrospect(token string) (*token.Introspection, error)
 }
 
 // New returns a new Server.
-func New(dexServerAddr string, cache cacheGetter, roleScopes map[string][]string) *Server {
+func New(c *config.Config, cache cacheGetter, roleScopes map[string][]string) *Server {
```

Comment:
Slightly prefer making the function take `tokenIntrospector` instead of `config.Config`.

Reply:
Let me handle this in a follow-up PR I have.